File: address-validation-us-3-dot-net-examples/GetSecondaryNumbersRestSdkExample.cs
using address_validation_us_3_dot_net.REST;

namespace address_validation_us_3_dot_net_examples
{
    internal static class GetSecondaryNumbersRestSdkExample
    {
        public static void Go(string LicenseKey, bool IsLive)
        { 
            Console.WriteLine("\r\n--------------------------------------------------------");
            Console.WriteLine("Address Validation US 3 - GetSecondaryNumbers - REST SDK");
            Console.WriteLine("--------------------------------------------------------");

            GetSecondaryNumbersClient.GetSecondaryNumbersInput getSecondaryNumbersInput = new(
                "136 West Canon Perdido St",
                "Santa Barbara",
                "CA",
                "93101",
                LicenseKey,
                IsLive
            );
            Console.WriteLine("\r\n* Input *\r\n");

            Console.WriteLine($"Address    : {getSecondaryNumbersInput.Address}");
            Console.WriteLine($"City       : {getSecondaryNumbersInput.City}");
            Console.WriteLine($"State      : {getSecondaryNumbersInput.State}");
            Console.WriteLine($"ZIP Code   : {getSecondaryNumbersInput.PostalCode}");
            Console.WriteLine($"Is Live    : {getSecondaryNumbersInput.IsLive.ToString()}");
            Console.WriteLine($"License Key: {getSecondaryNumbersInput.LicenseKey}");


            GSNResponse response = GetSecondaryNumbersClient.Invoke(getSecondaryNumbersInput);
            if (response.Error is null)
            {
                Console.WriteLine("\r\n* Validation *\r\n");

                Console.WriteLine($"Address          : {response.Address1}");
                Console.WriteLine($"City             : {response.City}");
                Console.WriteLine($"State            : {response.State}");
                Console.WriteLine($"ZIP+4            : {response.Zip}");
                Console.WriteLine($"Total Unit Count : {response.TotalCount.ToString()}");
                Console.WriteLine($"Secondary Numbers:");


                foreach (string unit in response.SecondaryNumbers)
                {
                    Console.WriteLine($"                  {unit}");
                }
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");

                Console.WriteLine($"Error Type     : {response.Error.Type}");
                Console.WriteLine($"Error Type Code: {response.Error.TypeCode}");
                Console.WriteLine($"Error Desc     : {response.Error.Desc}");
                Console.WriteLine($"Error Desc Code: {response.Error.DescCode}");
            }
        }
    }
}

File: address-validation-us-3-python-examples/get_secondary_numbers_rest_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-validation-us-3-python/REST"))

from get_secondary_numbers_rest import get_secondary_numbers

def get_secondary_numbers_rest_sdk_go(license, is_live_key):
    print("\n" + "-" * 51)
    print("Address Validation US 3 - GetSecondaryNumbers - REST SDK")
    print("-" * 51)

    print("\n* Input *\n")
    print(f"Address     : 136 West Canon Perdido St")
    print(f"City         : Santa Barbara")
    print(f"State        : CA")
    print(f"ZIP Code     : 93101")
    print(f"Is Live      : {is_live_key}")
    print(f"License Key  : {license}")

    try:
        response = get_secondary_numbers(
            address="136 West Canon Perdido St",
            city="Santa Barbara",
            state="CA",
            postal_code="93101",
            license_key=license, 
            is_live=True
        )

        if not response.get('Error'):
            print("\n* Validation *\n")
            print(f"Address 1          : {response.get('Address1', '')}")
            print(f"City               : {response.get('City', '')}")
            print(f"State              : {response.get('State', '')}")
            print(f"ZIP+4              : {response.get('Zip', '')}")
            print(f"Total Count        : {response.get('TotalCount', '')}")
            print(f"SecondaryNumbers   : {', '.join(response.get('SecondaryNumbers', []))}")
        else:
            print("\n* Error *\n")
            err = response['Error']
            print(f"Error Type     : {err.get('Type', '')}")
            print(f"Error Type Code: {err.get('TypeCode', '')}")
            print(f"Error Desc     : {err.get('Desc', '')}")
            print(f"Error Desc Code: {err.get('DescCode', '')}")
    except RuntimeError as e:
        print("Error calling service:", e)
